### DISCUSSION TITLE
How to make sure the user receives lightning payment?

### DISCUSSION THREAD

**Discussion by basharov** [lightningnetwork/lnd#7891] [Q&A]
I am working on a side project to allow users to receive some payments from it. When a user clicks a button to withdraw to a lightning address, my webapp makes a request to the host of that lightning address and makes a payment.

I can see the transaction has finished and everything is fine.

But I face situations when people say they didn't receive anything even though the logs show that it should have been received and all settled.

I know it can be some shady users actually receiving funds but claiming they didn't to try to get more from me. On the other hand, it can be that users have some technical issues or lack of experience using lightning and they really didn't receive anything.

So, I cannot say with confidence when it's a case with fraud and when it's really some issues on their side.

Is there a way to protect and to know if users try to cheat on me or if they need help?

**Comment by ffranr** [comment]
Hello @basharov ,

The user's node generated an invoice to receive payment from your node. In order to claim the funds that you are attempting to send, the **recipient node must reveal (send)** a "pre-image" to your node.

By revealing the pre-image, the recipient proves that they have received the payment. The pre-image can be used to verify that the hash lock in the payment matches the pre-image hash in the invoice. This serves as cryptographic proof that the payment was successfully made, and the recipient can trust that they have received the funds.

If the payment was successful then your node and the receiving node should both have access to the invoice and the pre-image. If you're node can reveal the pre-image then you can prove to the receiver that the payment was successful.

(A  successful payment means that control over some funds were handed over to the receiving node.)

I hope that's clear, let me know if it isn't.

**Comment by mxicoderspl** [comment]
Hello,
 I faced the same issue One user complained that he didn't receive the satoshis in his Blixt Wallet. He saw the transaction in the transactions list as open and after some time when he refreshed the transaction was also wiped out. So I checked our application and the transaction went well without any error. but when I actually checked in the RTL I didn't see that transaction and that amount was also not sent. When I checked the QR code generated from our application for receiving and scanned that again it showed that Already Withdrawn. Also, we had many successful transactions without any problems on the same application in the same interval but had issue with this single transaction.
So, What can be the possible cause behind it? and if there is any possibility of checking the logs in our node server related to this transaction.

Thank You.

**Reply by bitromortac** [reply]
You would need the payment hash to debug things. If you have that, you could look up whether you find it in `lncli listpayments`. Another way to check from logs whether the payment happened is to look for `[INF] SRVR: Adding preimage=PREIMAGE to witness cache for PAYMENT_HASH`. Perhaps related: https://github.com/lightningnetwork/lnd/issues/7860
